packages/zendesk-adapter/src/errors.ts

```diff
@@ -17,14 +17,87 @@ import _ from 'lodash'
 import { EOL } from 'os'
 import { client as clientUtils } from '@salto-io/adapter-components'
 import { safeJsonStringify } from '@salto-io/adapter-utils'
+import { logger } from '@salto-io/logging'
+
+const log = logger(module)
+
+type Error403 = {
+  errors: { title: string; detail: string }[]
+}
+
+type Error422 = {
+  description: string
+  details: Record<string, { description: string }[]>
+}
+
+type Error400 = {
+  error: {
+    title: string
+    message: string
+  }
+}
+
+const is403Error = (error: Record<string, unknown>): error is Error403 =>
+  (_.isArray(error.errors)
+    && (error.errors[0].title !== undefined)
+    && (error.errors[0].detail !== undefined))
+
+const is422Error = (error: Record<string, unknown>): error is Error422 =>
+  ((error.description !== undefined)
+    && (_.isPlainObject(error.details))
+    && (_.isObject(error.details))
+    && Object.values(error.details).every(val => _.isArray(val) && (val[0].description !== undefined)))
+
+const is400Error = (error: Record<string, unknown>): error is Error400 =>
+  (_.isPlainObject(error.error)
+    && (_.isObject(error.error))
+    && (error.error !== undefined)
+    && ('title' in error.error)
+    && ('message' in error.error))
+
+const error403ToString = (error: Error403): string => {
+  const errorArray = ['\nError details:']
+  error.errors.forEach(err => {
+    errorArray.push(`* Title: ${err.title}\n  Detail: ${err.detail}\n`)
+  })
```

Comment:
we prefer functional style - this can be
```suggestion
  const errorStrings = error.errors.flatMap(err => ([`* Title: ${err.title}`,`  Detail: ${err.detail}`]))
```